ext/mysqli/mysqli.c

```diff
@@ -1200,15 +1200,18 @@ void php_mysqli_fetch_into_hash(INTERNAL_FUNCTION_PARAMETERS, int override_flags
 			fci.param_count = 0;
 			fci.named_params = NULL;
 
-			if (ctor_params && Z_TYPE_P(ctor_params) != IS_NULL) {
+			if (ctor_params) {
 				if (zend_fcall_info_args(&fci, ctor_params) == FAILURE) {
 					/* Two problems why we throw exceptions here: PHP is typeless
 					 * and hence passing one argument that's not an array could be
 					 * by mistake and the other way round is possible, too. The
 					 * single value is an array. Also we'd have to make that one
 					 * argument passed by reference.
 					 */
-					zend_argument_error(zend_ce_exception, 3, "must be of type array, %s given", zend_zval_type_name(ctor_params));
+					zend_argument_error(zend_ce_exception, ERROR_ARG_POS(3),
+						"must be of type array when the specified class (%s) has a constructor",
```

Comment:
Is that quite correct? If I'm reading the code right, you can also pass null in that case, which will be interpreted as `[]`. I would just say `must of type ?array, %s given` for this branch.

Ideally, we would not accept null at all and allow an empty array for no constructor ... but the current null convention seems to be used quite consistently across DB extensions, so let's leave it as is.

Reply:
Now, that `ctor_params` became `a!`, this branch shouldn't allow a `null` value to enter, right?

> I would just say must of type ?array, %s given for this branch.

I think this error message wouldn't make much sense, because the value can already be either an array or null at this point.

Reply:
Right ... and that means this is actually dead code. You can replace the whole branch with a `ZEND_UNREACHABLE()`.

Reply:
Ah! Is this because `zend_fcall_info_args` can't fail in this case, isn't it? OK, I'll remove this a bit later.